provider/datagen/src/transform/segmenter/mod.rs

```diff
@@ -81,561 +81,542 @@ struct SegmenterRuleTable {
 }
 
 #[cfg(any(feature = "use_wasm", feature = "use_icu4c"))]
-impl crate::DatagenProvider {
-    fn generate_rule_break_data(&self, rules: &str) -> RuleBreakDataV1<'static> {
-        let segmenter: SegmenterRuleTable =
-            toml::from_str(rules).expect("The data should be valid!");
-
-        let data = maps::load_word_break(self).expect("The data should be valid!");
-        let wb = data.as_borrowed();
-
-        let data = maps::load_grapheme_cluster_break(self).expect("The data should be valid!");
-        let gb = data.as_borrowed();
-
-        let data = maps::load_sentence_break(self).expect("The data should be valid!");
-        let sb = data.as_borrowed();
-
-        let data = maps::load_line_break(self).expect("The data should be valid!");
-        let lb = data.as_borrowed();
-
-        let data = maps::load_east_asian_width(self).expect("The data should be valid!");
-        let eaw = data.as_borrowed();
-
-        let data = maps::load_general_category(self).expect("The data should be valid!");
-        let gc = data.as_borrowed();
-
-        let data = maps::load_script(self).expect("The data should be valid");
-        let script = data.as_borrowed();
-
-        let data = sets::load_extended_pictographic(self).expect("The data should be valid!");
-        let extended_pictographic = data.as_borrowed();
-
-        fn set_break_state(
-            break_state_table: &mut [i8],
-            property_length: usize,
-            left_index: usize,
-            right_index: usize,
-            break_state: i8,
-        ) {
-            let index = left_index * property_length + right_index;
-            if break_state_table[index] == UNKNOWN_RULE
-                || break_state_table[index] == NOT_MATCH_RULE
-            {
-                break_state_table[index] = break_state;
-            }
+fn generate_rule_break_data(
```

Comment:
nit: document that this is currently always called with a different provider that uses hardcoded data

Reply:
Is this blocking? I think that's fairly obvious from the single call site.

Reply:
I had to dig a bit. Not blocking